contracts/access/AccessControlDefaultAdminRules.sol

```diff
@@ -136,7 +136,7 @@ abstract contract AccessControlDefaultAdminRules is IAccessControlDefaultAdminRu
      * @dev See {AccessControl-_revokeRole}.
      */
     function _revokeRole(bytes32 role, address account) internal virtual override {
-        if (role == DEFAULT_ADMIN_ROLE) {
+        if (role == DEFAULT_ADMIN_ROLE && account == _currentDefaultAdmin) {
```

Comment:
Why not do 
```
bool revoked = super._revokeRole(role, account);
if (role == DEFAULT_ADMIN_ROLE && revoked) {
  delete _currentDefaultAdmin;
}
```
?

(feels better "integrated" to me)

Reply:
Both options work for me.

Reply:
I honestly don't see how this might work. `super._revokeRole()` doesn't return anything.

Reply:
> I honestly don't see how this might work. `super._revokeRole()` doesn't return anything.

Indeed ... I got confused with `EnumerableSet._add` and `EnumerableSet._remove` that return bool.
In 5.0 maybe we want this functions to return a bool